### PR TITLE
Added filters and sorting options for job list, added tooltip for tasks filters

### LINKED ISSUE #2959: Filters and sorting for jobs and tasks
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/opencv/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

We are running a large project in CVAT with a lot of Tasks and Jobs. It is very difficult to navigate over them. If the Jobs/Tasks list could be filtered and sorted (by assignee, reviewer, status, etc.), grouped by same parameters, it would be very convenient.

### Expected Behaviour
Get the ability to filter and sort Jobs/Tasks by assignee, reviewer, status, etc.

### Current Behaviour
No any filtering, sorting mechanism in CVAT.
### Possible Solution
Add filters and sorting to CVAT Jobs/Tasks list.

### Context

It is difficult to navigate if there are many Jobs /Tasks in the list. This feature can help.
### Your Environment
- Git hash commit (`git log -1`): commit 7a2dcfff61626dbfdaa694127fc1cff8ee10e000 (HEAD -> develop, origin/develop, origin/HEAD)
Author: Anastasia Yasakova <anastasia.yasakova@intel.com>
Date:   Tue Mar 9 10:41:33 2021 +0300
- Docker version `docker version` (e.g. Docker 17.0.05):  20.10.5
- Are you using Docker Swarm or Kubernetes? No
- Operating System and version (e.g. Linux, Windows, MacOS): GNU/Linux Ubuntu
4.15.0-136-generic
x86_64

### PR DESCRIPTION
**Pull request by bsekachev**
### Motivation and context
Resolved #2959
Screenshot from 2021-03-30 10-17-58

Screenshot from 2021-03-30 10-17-47

### How has this been tested?
Manual testing

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by bsekachev** [comment]
Hi @dvkruchinin 

We need some tests to cover different cases of using filters and sorting for job list:
- be sure that filter lists contain all the possible options
- try to combine different filters (for example reviewer + assignee + status)
- try sortings
- try to combine filters and sortings
- etc.

**Comment by dvkruchinin** [comment]
Hi @bsekachev 

> We need some tests to cover different cases of using filters and sorting for job list:

Such a test will be prepared.

File: cvat-ui/src/components/tasks-page/top-bar.tsx [hunk 1]
```diff
@@ -25,13 +28,71 @@ export default function TopBarComponent(props: VisibleTopBarProps): JSX.Element
...
                 <Col md={11} lg={9} xl={8} xxl={7}>
                     <Text className='cvat-title'>Tasks</Text>
-                    <Input.Search
-                        className='cvat-task-page-search-task'
-                        defaultValue={searchValue}
-                        onSearch={onSearch}
-                        size='large'
-                        placeholder='Search'
-                    />
+                    <CVATTooltip
```
Comment (ActiveChooN) [inline cvat-ui/src/components/tasks-page/top-bar.tsx:31]: Can you add this tooltip to project search field too?

Comment (bsekachev) [reply cvat-ui/src/components/tasks-page/top-bar.tsx:31]: Sure

Comment (bsekachev) [reply cvat-ui/src/components/tasks-page/top-bar.tsx]: Reworked

**Review by ActiveChooN — APPROVED** [review]

**Pull request merged**